### ISSUE TITLE
Language server does not respect client capabilities for signature help

### ISSUE THREAD

**Issue by puremourning** [microsoft/pyright#919]
**Describe the bug**

The language server protocol requires that the client declare support for "label offsets" in signature help:

```
	signatureInformation?: {
		/**
		 * Client capabilities specific to parameter information.
		 */
		parameterInformation?: {
			/**
			 * The client supports processing label offsets instead of a
			 * simple label string.
			 *
			 * @since 3.14.0
			 */
			labelOffsetSupport?: boolean;
		};
	};
```

However, the language sever here returns label offsets even when that support is not declared.

**To Reproduce**

Using the following LSP initialise:

```
2020-08-06 22:30:33,177 - DEBUG - TX: Sending message: b'Content-Length: 1154\r\n\r\n{"id":1,"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"experimental":{"statusNotification":true},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"documentationFormat":["plaintext","markdown"]},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"contentFormat":["plaintext","markdown"]},"signatureHelp":{"signatureInformation":{"documentationFormat":["plaintext","markdown"],"parameterInformation":{"labelOffsetSupport":false}}},"synchronization":{"didSave":true}},"workspace":{"applyEdit":true,"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]}},"workspaceEdit":{"documentChanges":true}}},"initializationOptions":{},"processId":32839,"rootPath":"/Users/ben/.vim/bundle/YouCompleteMe/third_party/ycmd","rootUri":"file:///Users/ben/.vim/bundle/YouCompleteMe/third_party/ycmd"}}'
```

specifically:

```
"signatureHelp":{"signatureInformation":{"documentationFormat":["plaintext","markdown"],"parameterInformation":{"labelOffsetSupport":false}}}
```

Send Signature help request:

```
2020-08-06 22:32:04,622 - DEBUG - TX: Sending message: b'Content-Length: 254\r\n\r\n{"id":2,"jsonrpc":"2.0","method":"textDocument/signatureHelp","params":{"position":{"character":23,"line":31},"textDocument":{"uri":"file:///Users/ben/.vim/bundle/YouCompleteMe/third_party/ycmd/ycmd/completers/language_server/generic_lsp_completer.py"}}}'
```

And received response:

```
2020-08-06 22:32:04,626 - DEBUG - RX: Received message: b'{"jsonrpc":"2.0","id":2,"result":{"signatures":[{"label":"(*args) -> None","parameters":[{"label":[1,6]}],"activeParameter":0}],"activeSignature":0,"activeParameter":0}}'
```

**Expected behavior**

The language server returns messages using only `string` value for the `label` key, per the specification:

```
/**
 * Represents a parameter of a callable-signature. A parameter can
 * have a label and a doc-comment.
 */
export interface ParameterInformation {

	/**
	 * The label of this parameter information.
	 *
	 * Either a string or an inclusive start and exclusive end offsets within its containing
	 * signature label. (see SignatureInformation.label). The offsets are based on a UTF-16
	 * string representation as `Position` and `Range` does.
	 *
	 * *Note*: a label of type string should be a substring of its containing signature label.
	 * Its intended use case is to highlight the parameter label part in the `SignatureInformation.label`.
	 */
	label: string | [number, number];

```

**Comment by erictraut** [comment]
Pyright is designed to work with VS Code. We don't guarantee that it works (or test it with) other language clients. I'll look at how straightforward this is to accommodate.

**Comment by puremourning** [comment]
Thanks. But a actually don’t spend time on it, I’m sure you have better things to do :). Well probably just implement the capability as it is fairly trivial our side ;)

You’re not the only server vendor that only tests with vscode :) I can understand why.

**Comment by erictraut** [comment]
This will be addressed in the next release of Pyright.

**Comment by puremourning** [comment]
woo thanks!

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.61, which I just published.
